websocket_message_test.go

```diff
@@ -76,231 +76,435 @@ func TestWebsocketMessage_Alias(t *testing.T) {
 func TestWebsocketMessage_CloseMessage(t *testing.T) {
 	reporter := newMockReporter(t)
 
-	msg := NewWebsocketMessage(reporter, websocket.CloseMessage, nil, 1000)
-
-	msg.CloseMessage()
-	msg.chain.assert(t, success)
-	msg.chain.clear()
-
-	msg.NotCloseMessage()
-	msg.chain.assert(t, failure)
-	msg.chain.clear()
-
-	msg.BinaryMessage()
-	msg.chain.assert(t, failure)
-	msg.chain.clear()
+	msg := NewWebsocketMessage(reporter, websocket.CloseMessage, nil, 10)
 
-	msg.NotBinaryMessage()
-	msg.chain.assert(t, success)
-	msg.chain.clear()
+	cases := []struct {
+		name   string
+		fn     func() *WebsocketMessage
+		testFn func() *WebsocketMessage
+		want   chainResult
+	}{
+		{
+			name:   "close message type with CloseMessage function",
+			fn:     msg.CloseMessage,
+			testFn: msg.CloseMessage,
+			want:   success,
+		},
+		{
+			name:   "close message type with NotCloseMessage function",
+			fn:     msg.CloseMessage,
+			testFn: msg.NotCloseMessage,
+			want:   failure,
+		},
+		{
+			name:   "close message type with BinaryMessage function",
+			fn:     msg.CloseMessage,
+			testFn: msg.BinaryMessage,
+			want:   failure,
+		},
+		{
+			name:   "close message type with NotBinaryMessage function",
+			fn:     msg.CloseMessage,
+			testFn: msg.NotBinaryMessage,
+			want:   success,
+		},
+		{
+			name:   "close message type with TextMessage function",
+			fn:     msg.CloseMessage,
+			testFn: msg.TextMessage,
+			want:   failure,
+		},
+		{
+			name:   "close message type with NotTextMessage function",
+			fn:     msg.CloseMessage,
+			testFn: msg.NotTextMessage,
+			want:   success,
+		},
+	}
 
-	msg.TextMessage()
-	msg.chain.assert(t, failure)
-	msg.chain.clear()
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.testFn().chain.assert(t, tc.want)
+			msg.chain.clear()
+		})
+	}
 
-	msg.NotTextMessage()
-	msg.chain.assert(t, success)
-	msg.chain.clear()
+	cases2 := []struct {
```

Comment:
Can we move each group of cases (cases, cases2, cases3) into a subtest, ideally with a meaningful name? (so it will be three subtests, each one with a table with sub-subtests).

Reply:
Same for other tests.

Reply:
done. But i divided case 1 into multiple subtests since there are different methods are used in case 1.